safekeeper/src/lib.rs

```diff
@@ -95,3 +96,11 @@ impl SafeKeeperConf {
         }
     }
 }
+
+/// Get unique thread id (Rust internal), with ThreadId removed for shorter printing
+pub fn get_tid() -> u64 {
```

Comment:
Hm, or I'm wrong?  Will it cache the value instead? Still I think if you need the pid the better option would be to use corresponding functions from nix crate ([gettid](https://docs.rs/nix/latest/nix/unistd/fn.gettid.html)?) since we already have it as a dependency.

Reply:
fixed, see discussion aove